app/components/post_draft/draft_input/draft_input.js

```diff
@@ -305,6 +310,14 @@ export default class DraftInput extends PureComponent {
 
         this.setInputValue('');
         this.input.current.changeDraft('');
+
+        if (data.goto_location) {
+            this.props.handleGotoLocation(data.goto_location, this.context.intl);
+        }
+    };
+
+    handleClosePermalink = () => {
```

Comment:
There is already an action for this https://github.com/mattermost/mattermost-mobile/blob/master/app/actions/views/permalink.ts#L46